tests/test_packages/functional/package_test/__init__.py

```diff
@@ -0,0 +1,273 @@
+# coding=utf-8
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+"""Setup for the package tests."""
+
+import json
+import multiprocessing
+import os
+import subprocess
+import sys
+import time
+import uuid
+
+sys.path.append(os.path.abspath(os.environ['TEST_TESTS_DIR']))
+from test_utils import get_free_port
+
+__PKG_ROOT = os.path.abspath(os.environ['TEST_CODECHECKER_DIR'])
+__LAYOUT_FILE_PATH = os.path.join(__PKG_ROOT, 'config', 'package_layout.json')
+with open(__LAYOUT_FILE_PATH) as layout_file:
+    __PACKAGE_LAYOUT = json.load(layout_file)
+sys.path.append(os.path.join(
+    __PKG_ROOT, __PACKAGE_LAYOUT['static']['codechecker_gen']))
+
+# stopping event for CodeChecker server
+__STOP_SERVER = multiprocessing.Event()
+
+
+def setup_package():
+    """Setup the environment for the tests. Check the test project twice,
+    then start the server."""
+    pkg_root = os.path.abspath(os.environ['TEST_CODECHECKER_DIR'])
+
+    env = os.environ.copy()
+    env['PATH'] = os.path.join(pkg_root, 'bin') + ':' + env['PATH']
+
+    tmp_dir = os.path.abspath(os.environ['TEST_CODECHECKER_PACKAGE_DIR'])
+    workspace = os.path.join(tmp_dir, 'workspace')
+    if not os.path.exists(workspace):
+        os.makedirs(workspace)
+
+    test_project_path = os.path.join(
+        os.path.abspath(os.environ['TEST_TESTS_DIR']),
+        'test_projects',
+        'test_files')
+
+    clang_version = os.environ['TEST_CLANG_VERSION']
+
+    database = {

```

Comment:
How about renaming database to pg_database_config/pg_db_config? The same way as test_config?